core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -302,12 +302,12 @@ class DAGScheduler(
       shuffleDep: ShuffleDependency[_, _, _],
       firstJobId: Int): ShuffleMapStage = {
     val rdd = shuffleDep.rdd
-    val numTasks = rdd.partitions.size
+    val numTasks = rdd.partitions.length
```

Comment:
I'm not sure how true it is with more recent versions of Scala, but there at least was a time when `Array#size` didn't perform nearly as well as `Array#length`.